File: Python/Algorithm/Algorithm I/191. Number of 1 Bits.py
class Solution:
    def hammingWeight(self, n: int) -> int:
        """
        Count the number of '1' bits in the binary representation of an integer.

        Args:
            n (int): An integer.

        Returns:
            int: The number of '1' bits.

        Time Complexity: o(n) where n is the number of bits in the binary representation of the integer.
        Space Complexity: o(1) since we are not using any additional space.
        """
        oneBitCounter = 0

        while n > 0:
            LSB_Value = n & 1
            oneBitCounter += LSB_Value

            n >>= 1

        return oneBitCounter

File: Java/Algorithm/Algorithm I/191. Number of 1 Bits.java
public class Solution {

  public int hammingWeight(int n) {
    int oneBitCounter = 0;

    for (int iteration = 0; iteration < 32; iteration++) {
      int LSB_Value = n & 1;
      oneBitCounter += LSB_Value;

      n >>= 1;
    }

    return oneBitCounter;
  }
}
